src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,329 @@
+""" Automation of memory cache suite
+
+"""
+from __future__ import print_function
+import time
+import pytest
+import subprocess
+from sssd.testlib.common.utils import sssdTools, LdapOperations
+from constants import ds_instance_name
+
+
+def find_logs(multihost, log_name, string_name):
+    "This function will find strings in a log file"
+    log_str = multihost.client[0].get_file_contents(log_name).decode('utf-8')
+    assert string_name in log_str
+
+
+def clear_only_domain_log(multihost):
+    "This function will clear domain logs"
+    client = multihost.client[0]
+    log_ssd = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+    client.run_command(f'> {log_ssd}')
+
+
+@pytest.mark.tier1_4
+@pytest.mark.usefixtures('setup_sssd', 'create_posix_usersgroups')
+@pytest.mark.offline
+class TestPasswordPolicy(object):
+    """
+    This is test case class for sssd memory cache suite
+    """
+    @staticmethod
+    def test_bz789507(multihost):
+        """
+        :title: SSSD should provide fast in memory cache bz789507
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=789507
+        :id: 19dfa7ae-96e9-11ed-adf3-845cf3eff344
+        :steps:
+            1. Configure sssd nss
+            2. Check SSSD should provide fast in memory cache
+                to provide similar functionality as NSCD currently provides
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'filter_groups': 'root',
```

Comment:
You are changing the config, backupsssdconf fixture should be used to restore the config.

Reply:
Class fixture backupsssd_conf is being used, it will take back up of sssd.conf before this test and will restore after the tests finises

Reply:
I think that class fixture will not cleanup state/changes between tests so changes to config done in one test will leak to the following ones.